File: frontend/src/types/task.ts
export interface Task {
  id: number;
  title: string;
  description: string | null;
  completed: boolean;
}

export interface TaskCreate {
  title: string;
  description?: string | null;
  completed?: boolean;
}

export interface TaskUpdate {
  title: string;
  description?: string | null;
  completed: boolean;
}

export interface TaskPartialUpdate {
  title?: string;
  description?: string | null;
  completed?: boolean;
}

File: backend/app/schemas/task.py
from pydantic import BaseModel, Field
from typing import Optional

class TaskBase(BaseModel):
    title: str = Field(..., min_length=1, max_length=200)
    description: Optional[str] = None
    completed: bool = False

class TaskCreate(TaskBase):
    pass

class TaskUpdate(TaskBase):
    pass

class TaskPartialUpdate(BaseModel):
    title: Optional[str] = Field(None, min_length=1, max_length=200)
    description: Optional[str] = None
    completed: Optional[bool] = None

class TaskResponse(TaskBase):
    id: int

    class Config:
        from_attributes = True
